ui-tests/src/test/resources/features/integrations/amq-to-rest.feature

```diff
@@ -68,20 +68,19 @@ Feature: Integration - AMQ to REST
     When navigate to the "Integrations" page
     Then wait until integration "AMQ to TODO integration" gets into "Running" state
 
-    # [WIP] - workaround for now due to https://github.com/syndesisio/todo-example/issues/16
-    #  test should go to the Todo app and use JMS form instead of code (for real quickstart testing)
-    #  e.g. When navigate to Todo app ; Then check Todo list grows in "15" second
+    And navigate to Todo app
+    And publish JMS message on Todo app page from resource "tutorialAmqToRestMessage1.xml"
 
-    And publish JMS message from resource "tutorialAmqToRestMessage1.xml" to "queue" with name "inventoryReceived"
-    And checks that query "SELECT * FROM todo WHERE task LIKE '%Contact Joe Doe%ABC789%'" has "1" output
-    And checks that query "SELECT * FROM todo WHERE task LIKE '%Contact John Smith%XYZ123%'" has no output
+    And check that "1". task on Todo app page contains text "Task: Contact Joe Doe, 987 654 321. Damaged items: ABC789."
+    And check Todo list has "1" items
 
-    And publish JMS message from resource "tutorialAmqToRestMessage2.xml" to "queue" with name "inventoryReceived"
-    And checks that query "SELECT * FROM todo WHERE task LIKE '%Contact Joe Doe%ABC789%'" has "2" output
-    And checks that query "SELECT * FROM todo WHERE task LIKE '%Contact John Smith%XYZ123%Contact Joe Doe%ABC789%'" has "1" output
+    And publish JMS message on Todo app page from resource "tutorialAmqToRestMessage2.xml"
+    And check that "1". task on Todo app page contains text "Task: Contact Joe Doe, 987 654 321. Damaged items: ABC789."
+    And check that "2". task on Todo app page contains text "Task: Contact John Smith, 123 456 789. Damaged items: XYZ123. Contact Joe Doe, 987 654 321. Damaged items: ABC789."
+    And check Todo list has "2" items
 
-    And publish JMS message from resource "tutorialAmqToRestMessage3.xml" to "queue" with name "inventoryReceived"
-    And checks that query "SELECT * FROM todo WHERE task LIKE '%Contact Joe Doe%ABC789%'" has "2" output
-    And checks that query "SELECT * FROM todo WHERE task LIKE '%Contact John Smith%XYZ123%Contact Joe Doe%ABC789%'" has "1" output
-    And checks that query "SELECT * FROM todo WHERE task LIKE '%No contact found%ABC789%'" has "1" output
-    And checks that query "SELECT * FROM todo " has "3" output
+    And publish JMS message on Todo app page from resource "tutorialAmqToRestMessage3.xml"
+    And check that "1". task on Todo app page contains text "Task: Contact Joe Doe, 987 654 321. Damaged items: ABC789."
+    And check that "2". task on Todo app page contains text "Task: Contact John Smith, 123 456 789. Damaged items: XYZ123. Contact Joe Doe, 987 654 321. Damaged items: ABC789."
+    And check that "3". task on Todo app page contains text "Task: No contact found. Damaged items: ABC789."
+    And check Todo list has "3" items
```

Comment:
It might be better to check the total count first ... when failing, you would know that the number of rows is wrong. In current shape you might suspect that the first item contents are wrong. ....

Reply:
Good point, fixed